docs/source/dynamo/custom-backends.rst

```diff
@@ -1,8 +1,95 @@
 Custom Backends
 ===============
 
+Overview
+--------
+
+``torch.compile`` provides a straightforward method to enable users
+to define custom backends.
+
+A backend function has the contract
+``(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]) -> Callable``.
+
+Backend functions can be called by TorchDynamo, the graph tracing component of ``torch.compile``,
+after tracing an FX graph and are
+expected to return a compiled function that is equivalent to the traced FX graph.
+The returned callable should have the same contract as the ``forward`` function of an ``nn.Module``:
+``(*args: Any) -> Any``.
+
+In order for TorchDynamo to call your backend, pass your backend function as the ``backend`` kwarg in
+``torch.compile``. For example,
+
+.. code-block:: python
+
+    import torch
+
+    def my_custom_backend(gm, example_inputs):
+        return gm.forward
+
+    def f(...):
+        ...
+
+    f_opt = torch.compile(f, backend=my_custom_backend)
+
+    @torch.compile(backend=my_custom_backend)
+    def g(...):
+        ...
+
+See below for more examples.
+
+Registering Custom Backends
+---------------------------
+
+You can register your backend using the ``register_backend`` decorator, for example,
+
+.. code-block:: python
+
+    from torch._dynamo.optimizations import register_backend
+
+    @register_backend
+    def my_compiler(gm, example_inputs):
+        ...
+
+Registration serves two purposes:
+
+* You can pass a string containing your backend function's name to ``torch.compile`` instead of the function itself,
+  for example, ``torch.compile(model, backend="my_compiler")``.
+* It is required for use with the `minifier <https://pytorch.org/docs/master/dynamo/troubleshooting.html>`__. Any generated
+  code from the minifier must call your code that registers your backend function, typically through an ``import`` statement.
+
+Custom Backends after AOTAutograd
+---------------------------------
+
+If you want your backend to be called by AOTAutograd rather than TorchDynamo, you can wrap your backend with
```

Comment:
Add a sentence about why someoneone would want this (support training, normalized IRs)